components/InputBox/index.tsx

```diff
@@ -144,22 +152,63 @@ const InputBox = ({
 
       // -- Code for local message handling ended
 
-      let payload = {
-        chatroom_id: chatroomID,
-        created_at: new Date(Date.now()),
-        has_files: false,
-        text: message.trim(),
-        // attachment_count?: any;
-        replied_conversation_id: replyMessage?.id,
-      };
-      let response = await dispatch(onConversationsCreate(payload) as any);
-      // addItem(payload);
-      // if (!!response) {
-      //   setMessage('');
-      // }
+      // condition for request DM for the first time
+      if (chatroomType === 10 && chatRequestState === null) {
+        let response = await myClient.requestDmAction({
+          chatroom_id: chatroomID,
+          chat_request_state: 0,
+          text: message.trim(),
+        });
+
+        dispatch({
+          type: SHOW_TOAST,
+          body: {isToast: true, msg: 'Direct messaging request sent'},
+        });
+
+        //dispatching redux action for local handling of chatRequestState
+        dispatch({
+          type: UPDATE_CHAT_REQUEST_STATE,
+          body: {chatRequestState: 0},
+        });
+        console.log('response requestDmAction ==', response);
+      } else {
+        let payload = {
+          chatroom_id: chatroomID,
+          created_at: new Date(Date.now()),
+          has_files: false,
+          text: message.trim(),
+          // attachment_count?: any;
+          replied_conversation_id: replyMessage?.id,
+        };
+        let response = await dispatch(onConversationsCreate(payload) as any);
+      }
     }
   };
 
+  // function calls a confirm alert which will further call onSend function onConfirm.
+  const sendDmRequest = () => {
+    Alert.alert(
+      'Send DM request?',
```

Comment:
can we use any string constants/resources, to store all these messages?, so that they are easily change?